cmd/choice_flag.go

```diff
@@ -0,0 +1,93 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"strings"
+
+	"github.com/urfave/cli/v2"
+)
+
+// ChoiceFlag is a cli.Flag whose value is populated from preconfigured choices
+type ChoiceFlag struct {
```

Comment:
I'm really surprised that the StringFlag type doesn't support this out of the box 🙂

Reply:
Just to clarify: Yes, I'm actually surprised that cli.Flag doesn't have such a functionality 🙂 There is a way to get around that using the `Action` field where you can add custom validation logic when you use, for instance, a StringFlag, but that's it. Having that in separate struct (like your `ChoiceFlag`) is the by-far nicer solution 👍

Reply:
I worked with Ricky on adding the ChoiceValue type, which satisfies `flag.Value` so that we could do the validation after `flagset.Parse` was called.

We unfortunately weren't able to do the validation in Apply, but that's OK; we think the ChoiceValue is a pretty clean solution.